### ISSUE TITLE
--framework=v4.0 result in "Unknown runtime type: Any'

### ISSUE THREAD

**Issue by tgaertner-recom** [nunit/nunit-console#1165]
I updated the console runner in our CI because i tried to circumvent another issue, where the runner unexpectedly exits with -1 and without a single message or result summary. In turn I also removed --inprocess argument. The --framework argument was set to v4.0.30319. That was quite fine for a long while.

I got the code myself to get some hints what "runtime type any" means and why it should not possible to execute with any framework present: 
* There's a switch in AgentProcessor.GetTestAgentExePath that does not handle RuntimeType.Any... 
* RuntimeFramework.Parse returns that Any for the provided "v4.0.30319" (which was a fine argument until now)

I would appreciate some better error messages regarding this argument. I'm changing it to "net-4.5" regarding to the implementation of "RuntimeFramework.Parse". This fixes the behavior of the runner for me and it executes our tests isolated from each other. That in turn mitigates the strange exitcode -1 issue I was faced at the beginning of this odyssey.

```
[Runtests] ErrorLevel before NUnit: 0

c:\Builds\ci\pipeline>nunit3-console.exe [full load of assemblies] --x86 --trace=Debug --framework=v4.0.30319 --result=test_results_fast.xml --timeout=15000 

NUnit Console 3.99.0.0-VSIDE (Debug)
Copyright (c) 2022 Charlie Poole, Rob Prouse
Montag, 28. Februar 2022 12:18:14

Runtime Environment
   OS Version: Microsoft Windows NT 6.2.9200.0
   Runtime: .NET Framework CLR v4.0.30319.42000

Test Files
 [full load of assemblies]

Failures and Warnings

1) Error : ********.dll
System.InvalidOperationException : Der Prozess kann nicht gestartet werden, da ein Dateiname nicht angegeben wurde.
--InvalidOperationException
Der Prozess kann nicht gestartet werden, da ein Dateiname nicht angegeben wurde.
   bei System.Diagnostics.Process.Start()
   bei NUnit.Engine.Services.TestAgency.GetAgent(TestPackage package) in C:\Users\tgaertner\Downloads\nunit-console-3.15.0\nunit-console-3.15.0\src\NUnitEngine\nunit.engine\Services\TestAgency.cs:Zeile 75.
   bei NUnit.Engine.Runners.ProcessRunner.CreateAgentAndRunnerIfNeeded() in C:\Users\tgaertner\Downloads\nunit-console-3.15.0\nunit-console-3.15.0\src\NUnitEngine\nunit.engine\Runners\ProcessRunner.cs:Zeile 250.
   bei NUnit.Engine.Runners.ProcessRunner.RunTests(ITestEventListener listener, TestFilter filter) in C:\Users\tgaertner\Downloads\nunit-console-3.15.0\nunit-console-3.15.0\src\NUnitEngine\nunit.engine\Runners\ProcessRunner.cs:Zeile 136.

[...]

Test Run Summary
  Overall result: Failed
  Test Count: 0, Passed: 0, Failed: 0, Warnings: 0, Inconclusive: 0, Skipped: 0
  Start time: 2022-02-28 11:18:14Z
    End time: 2022-02-28 11:18:14Z
    Duration: 0.118 seconds

Results (nunit3) saved as test_results_fast.xml
[Runtests] NUnit exit code is -100
```

```
12:51:41.999 Info  [10] ProcessRunner: Running **********.dll
...
12:51:42.002 Debug [10] AgentProcess: GetTestAgentExePath(v4.0, False)
...
12:51:42.002 Debug [10] AgentProcess: Checking for agents at C:\ProgramData\chocolatey\lib\nunit-console-runner\tools\agents
12:51:42.002 Error [10] AgentProcess: Unknown runtime type: Any
...
12:51:42.040 Error [10] ProcessRunner: Failed to run remote tests --InvalidOperationException
Diesem Objekt ist kein Prozess zugeordnet.
   bei System.Diagnostics.Process.EnsureState(State state)
   bei System.Diagnostics.Process.EnsureState(State state)
   bei System.Diagnostics.Process.get_Id()
   bei NUnit.Engine.Services.TestAgency.GetAgent(TestPackage package) in C:\Users\tgaertner\Downloads\nunit-console-3.15.0\nunit-console-3.15.0\src\NUnitEngine\nunit.engine\Services\TestAgency.cs:Zeile 75.
   bei NUnit.Engine.Runners.ProcessRunner.CreateAgentAndRunnerIfNeeded() in C:\Users\tgaertner\Downloads\nunit-console-3.15.0\nunit-console-3.15.0\src\NUnitEngine\nunit.engine\Runners\ProcessRunner.cs:Zeile 250.
   bei NUnit.Engine.Runners.ProcessRunner.RunTests(ITestEventListener listener, TestFilter filter) in C:\Users\tgaertner\Downloads\nunit-console-3.15.0\nunit-console-3.15.0\src\NUnitEngine\nunit.engine\Runners\ProcessRunner.cs:Zeile 136.
```

**Comment by CharliePoole** [comment]
The console runner header indicates you are running a version which you built under the VS IDE. Can you indicate what source code you are using? Also, based on the message line at 12:51:42.002, it appears that you installed that build under chocolatey. Could you explain a bit more about what you did?

**Comment by tgaertner-recom** [comment]
The original version (that I am also now back to) is 3.15 updated at the last friday. 

The version I used in between was a local clone of tag '3.15' with some manual debugging cues and PDBs to find the right spots for exceptions. I was just lazy and replaced the locally compiled files in the chocolatey directory. So I could easily return to the original build.

**Comment by CharliePoole** [comment]
If your problem is solved in 3.15, I'd like to close this. Runtime type `Any` is an artifact, which will disappear from the 4.0 build in the next week or so.

**Comment by tgaertner-recom** [comment]
That's OK for me. There is this issue here for documentation, whenever someone else encounters this problem again.
